src/components/ui/MonacoEditor.svelte

```diff
@@ -78,54 +81,84 @@
       ...(value !== null && { value }),
     };
     monaco = await import('monaco-editor');
-    monaco.languages.typescript.typescriptDefaults.setWorkerOptions({
-      customWorkerPath: '/customTS.worker.js',
-    });
     editor = monaco.editor.create(div, options, override);
 
-    editor.onDidChangeModelContent((e: Editor.IModelContentChangedEvent) => {
-      const newValue = editor?.getModel()?.getValue();
-      if (newValue !== undefined) {
-        dispatch('didChangeModelContent', { e, value: newValue });
-      }
-    });
-
-    // So.. there is no way to check when the model is initialized apparently!
-    // https://github.com/microsoft/monaco-editor/issues/115
-
-    // If we accidentally call the `getTypeScriptWorker()` function to early, it throws.
-    // Just use retry with exponential back-off to get it!
-    promiseRetry(
-      async () => {
-        let tsWorker: TypeScriptWorker | null = null;
-
-        // Errors in this block indicate failure to find a loaded worker
-        // so transform to the specific error type we care about.
-        try {
-          const getWorker: ((...uris: Uri[]) => Promise<TypeScriptWorker>) | undefined =
-            await monaco?.languages.typescript.getTypeScriptWorker();
-          if (getWorker) {
-            tsWorker = await getWorker();
-          }
-        } catch (e) {
-          throw new ShouldRetryError();
+    if (language && language === 'typescript') {
+      monaco.languages.typescript.typescriptDefaults.setWorkerOptions({
+        customWorkerPath: '/customTS.worker.js',
+      });
+
+      codeActionProvider = monaco.languages.registerCodeActionProvider(language, {
+        provideCodeActions: (model, range, context, token) => {
```

Comment:
Can you remove this unused param?

Reply:
I tried to remove the unused variables but they are needed by the `provideCodeAction` and causes the action to not register.

Reply:
Does the lint error still occur if you add an underscore to the parameter name?